### PR TITLE
Sushiswap connector

### PR DESCRIPTION
**Pull request by bhavik-m**
- [x] Add Sushiswap Connector 
    - [x] Mainnet
    - [x] Polygon
    - [x] Arbitrum
    - [x] Avalanche
 - [x] Deploy Sushiswap Connector 
    - [x] Mainnet
    - [x] Polygon
    - [x] Arbitrum
    - [x] Avalanche
- [x] Add to dsa-connect [here](https://github.com/Instadapp/dsa-connect/pull/121)

### CODE REVIEW

File: contracts/arbitrum/connectors/sushiswap/main.sol [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+        setUint(setId, _buyAmt);
+
+        _eventName = "LogSell(address,address,uint256,uint256,uint256,uint256)";
+        _eventParam = abi.encode(buyAddr, sellAddr, _buyAmt, _sellAmt, getId, setId);
+    }
+}
+
+contract ConnectV2SushiswapArbitrum is SushipswapResolver {
+    string public constant name = "Sushiswap-v1.1";
```
Comment (thrilok209) [inline contracts/arbitrum/connectors/sushiswap/main.sol]:
```suggestion
    string public constant name = "Sushiswap-v1";
```

File: contracts/avalanche/connectors/sushiswap/main.sol [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+        setUint(setId, _buyAmt);
+
+        _eventName = "LogSell(address,address,uint256,uint256,uint256,uint256)";
+        _eventParam = abi.encode(buyAddr, sellAddr, _buyAmt, _sellAmt, getId, setId);
+    }
+}
+
+contract ConnectV2SushiswapAvalanche is SushipswapResolver {
+    string public constant name = "Sushiswap-v1.1";
```
Comment (thrilok209) [inline contracts/avalanche/connectors/sushiswap/main.sol]:
```suggestion
    string public constant name = "Sushiswap-v1";
```

**Review by thrilok209 — APPROVED** [review]

**Review by pradyuman-verma — APPROVED** [review]

**Comment by bhavik-m** [comment]
### Deployed on Polygon:

**Name:** `SUSHISWAP-A`
**Address:** [0xF18d8635B2717A82891A0C52C7798065F7b9aA1D](https://polygonscan.com/address/0xF18d8635B2717A82891A0C52C7798065F7b9aA1D#code)
**tx:** [here]()

**Comment by bhavik-m** [comment]
### Deployed on Avalanche:

**Name:** `SUSHISWAP-A`
**Address:** [0x8dA60dee0815a08d16C066b07814b10722fA9306](https://snowtrace.io/address/0x8da60dee0815a08d16c066b07814b10722fa9306#code)
**tx:** [here](https://snowtrace.io/tx/0xfb25c0228e7e2e3a0a69f3dca6242a7a5083b0ff568d4d7e367186573f69e7f2)

**Comment by bhavik-m** [comment]
### Deployed on Arbitrum:

**Name:** `SUSHISWAP-A`
**Address:** [0x8a7fCeE0e1Ff6DB33C7E83060c18e3B97915A970](https://arbiscan.io/address/0x8a7fcee0e1ff6db33c7e83060c18e3b97915a970#code)
**tx:** [here](https://arbiscan.io/tx/0x5d4788fd0c0e42a547bfc77b71a071e4bf1bc065511f458a5bf624f4da5f54ec)

**Comment by bhavik-m** [comment]
### Deployed on Mainnet:

**Name:** `SUSHISWAP-A`
**Address:** [0x90D45f987bCad6DD7FB5fE0994124D455cf30Bca](https://etherscan.io/address/0x90D45f987bCad6DD7FB5fE0994124D455cf30Bca#code)
**tx** [here](https://etherscan.io/tx/0xdc5939cd380f6ee1ddd34d8d5e5b117ac519a77c2bdb114371ddc47b0b447ef9)

**Pull request merged**